src/org/zaproxy/zap/extension/AllInOneNotes/ExtensionAllInOneNotes.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.AllInOneNotes;
+
+import java.awt.CardLayout;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+
+import javax.swing.*;
+
+import org.apache.log4j.Logger;
+import org.jdesktop.swingx.JXTable;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.extension.AbstractPanel;
+import org.parosproxy.paros.extension.ExtensionAdaptor;
+import org.parosproxy.paros.extension.ExtensionHook;
+import org.parosproxy.paros.extension.ExtensionHookView;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.zaproxy.zap.utils.TableExportButton;
+import org.zaproxy.zap.view.ZapMenuItem;
+
+public class ExtensionAllInOneNotes extends ExtensionAdaptor {
+
+    private ZapMenuItem menuExample;
+    // The name is public so that other extensions can access it
+    public static final String NAME = "ExtensionAllInOneNotes";
+
+    // The i18n prefix, by default the package name - defined in one place to make it easier
+    // to copy and change this example
+    protected static final String PREFIX = "allInOneNotes";
+
+    /**
+     * Relative path (from add-on package) to load add-on resources.
+     * 
+     * @see Class#getResource(String)
+     */
+    private static final String RESOURCES = "resources";
+
+    private static final ImageIcon ICON = new ImageIcon(
+            ExtensionAllInOneNotes.class.getResource(RESOURCES + "/notepad.png"));
+
+    private AbstractPanel statusPanel;
+    private ListSelectionModel tableSelectionModel;
+
+    private static final Logger LOGGER = Logger.getLogger(ExtensionAllInOneNotes.class);
+    private static ExtensionHookView hookView;
+
+    private static JButton reload = new JButton("Reload Notes");
```

Comment:
The button name should come i18n from the Messages.properties file. 
Something like `new JButton(Constant.messages.getString("allInOneNotes.reload.button");` with Messages.properties containing: `allInOneNotes.reload.button=Reload Notes`

Reply:
agreed! changed